low-level/solution/lr2/server/server_resource_map.py

```diff
@@ -0,0 +1,84 @@
+#!/bin/python3
+
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com
+
+import re
+from cortx.utils.discovery.resource_map import ResourceMap
+from framework.utils.service_logging import init_logging
+from framework.utils.conf_utils import (SSPL_CONF, Conf, SYSTEM_INFORMATION,
+    LOG_LEVEL)
+
+
+class ServerResourceMap(ResourceMap):
+    """
+    ServerResourceMap class provides resource map and related information
+    like health, manifest, etc,.
+    """
+
+    name = "server"
+
+    def __init__(self):
+        """Initialize server."""
+        super().__init__()
+        logging_level = Conf.get(SSPL_CONF,
+            f"{SYSTEM_INFORMATION}>{LOG_LEVEL}", "INFO")
+        init_logging('resource', logging_level)
+
+    @staticmethod
+    def get_health_info(rpath):
+        """
+        Fetch health information for given resource path
+
+        rpath: Resource path to fetch its health
+               Examples:
+                    node>compute[0]
+                    node>compute[0]>hw
+                    node>compute[0]>hw>disks
+        """
+        from health import ServerHealth
+        health = ServerHealth()
+        info = health.get_data(rpath)
+        return info
+
+    @staticmethod
+    def get_manifest_info(rpath):
+        """
+        Fetch manifest information for given resource path
+
+        rpath: Resource path to fetch its manifest
+               Examples:
+                    node>compute[0]
+        """
+        from manifest import ServerManifest
+        manifest = ServerManifest()
+        info = manifest.get_data(rpath)
+        return info
+
+    @staticmethod
+    def get_node_details(node):
```

Comment:
Can we rename `get_node_details()` to `get_node_info()`
We are following the pattern `get_[xyz]_info()` for most of the functions so I think here also we should go with the similar function name.

Reply:
Done.